samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,268 @@
+|                 |             |             |             | Required    |
++-----------------+-------------+-------------+-------------+-------------+
+| Data Out        | M           | Read, Write |             | Encryption  |
+| - Client        |             |             |             | Required    |
+| Characteristic  |             |             |             |             |
+| Configuration   |             |             |             |             |
+| descriptor      |             |             |             |             |
++-----------------+-------------+-------------+-------------+-------------+
+
+Information
+-----------
+
+This characteristic is used to get Info message.
+when read, the Information characteristic gives an encoded Protocol Buffers message in type of Info defined in version.proto.
+It is used by client to determine which version of Wi-Fi provisioning service is running on this server.
+
+Operation Control Point
+-----------------------
+
+This characteristic is used to send Request message and get Response message.
+The client shall send an encoded message in type of Request defined in request.proto, and the server shall send an encoded message in type of Response defined in response.proto.
+
+Data Out
+--------
+
+This characteristic is used to send notifications to the client in format of Result message.
+
+Advertisement
+-------------
+
+After power up, the device will first check if it is provisioned or not.
+If so, it will load saved credentials and try to connect the AP specified in the credential.
+After 30 seconds (timeout will occur if the attempt fails), it will start BLE advertising.
+If it is not provisioned, it will advertise immediately.
+
+The advertising data contains several fields to facilitate possible provisioning attempt.
+It will contain the UUID of Wi-Fi provisioning service, and a 4-byte service data.
+
++----------+----------+-----------+----------+
+|  Byte 1  |  Byte 2  |   Byte 3  |  Byte 4  |
++==========+==========+===========+==========+
+| Version  |         Flag         |   RSSI   |
++----------+----------------------+----------+
+
+Version: 8-bit unsigned integer. The value is the version of Wi-Fi provisioning service.
+
+Flag: 16-bit little endian field. Byte 2 (first byte on air) is LSB, and Byte 3 (second byte on air) is MSB.
+
+- Bit 0: provisioning status. 1 if device is provisioned, 0 otherwise.
+
+- Bit 1: connection status. 1 if Wi-Fi is connected, 0 otherwise.
+
+- Bit 2 - 15: RFU
+
+RSSI: 8-bit signed integer. If Wi-Fi is connected, it is the signal strength. Otherwise, it is meaningless.
+
+Depending on the provisioning status, the advertising interval differs.
+If it is not provisioned, the interval is 100ms. If it is provisioned, the interval is 1 second.
+
+Configuration management
+========================
+
+Once the device receives Wi-Fi configuration, it should keep the data in proper location, and save or delete the record according to incoming command.
```

Comment:
Do we need to talk about what the device does since it's not strictly related to the sample?

We say component here.

I would rewrite this as:
The configuration management component stores and deletes Wi-Fi configurations according to incoming commands. It ensures that the data in RAM and flash are in sync, and exposes APIs to add or remove configurations in a multi-thread environment.